File: backend/app/schemas/article.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel

from app.schemas.source import SourceResponse


class VerificationInfo(BaseModel):
    source_name: str
    matched_title: str | None
    matched_url: str | None
    similarity_score: float | None

    model_config = {"from_attributes": True}


class ArticleListItem(BaseModel):
    id: UUID
    title: str
    summary: str | None
    image_url: str | None
    category_slug: str | None
    published_at: datetime | None
    is_verified: bool
    verification_count: int
    has_audio: bool = False

    model_config = {"from_attributes": True}


class ArticleDetail(BaseModel):
    id: UUID
    title: str
    summary: str | None
    content: str | None
    original_url: str | None
    image_url: str | None
    category_slug: str | None
    published_at: datetime | None
    is_verified: bool
    verification_count: int
    primary_source: SourceResponse | None
    verifications: list[VerificationInfo]
    has_audio: bool = False

    model_config = {"from_attributes": True}


class PaginatedArticles(BaseModel):
    items: list[ArticleListItem]
    total: int
    page: int
    page_size: int
    has_next: bool

File: frontend/src/types/article.ts
// ###########################################################################
// # Types Article — Interfaces TypeScript pour les articles
// # ArticleListItem: carte dans le feed (titre, resume, image, sources)
// # ArticleDetail: article complet avec contenu + verifications
// # PaginatedArticles: reponse paginee de l'API
// ###########################################################################

export interface ArticleListItem {
  id: string;
  title: string;
  summary: string | null;
  image_url: string | null;
  category_slug: string | null;
  published_at: string | null;
  is_verified: boolean;
  verification_count: number;
  has_audio: boolean;
}

export interface VerificationInfo {
  source_name: string;
  matched_title: string | null;
  matched_url: string | null;
  similarity_score: number | null;
}

export interface SourceInfo {
  id: number;
  name: string;
  url: string;
  source_type: string;
  reliability_tier: number;
  country_code: string | null;
}

export interface ArticleDetail {
  id: string;
  title: string;
  summary: string | null;
  content: string | null;
  original_url: string | null;
  image_url: string | null;
  category_slug: string | null;
  published_at: string | null;
  is_verified: boolean;
  verification_count: number;
  primary_source: SourceInfo | null;
  verifications: VerificationInfo[];
  has_audio: boolean;
}

export interface PaginatedArticles {
  items: ArticleListItem[];
  total: number;
  page: number;
  page_size: number;
  has_next: boolean;
}
